apps/api/src/app/layouts/e2e/delete-layout.e2e.ts

```diff
@@ -0,0 +1,69 @@
+import { UserSession } from '@novu/testing';
+import { expect } from 'chai';
+
+import { TemplateVariableTypeEnum } from '../types';
+
+const BASE_PATH = '/v1/layouts';
+
+describe('Delete a layout - /layouts/:layoutId (DELETE)', async () => {
+  let session: UserSession;
+
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
+  it('should soft delete the requested layout successfully if exists in the database for that user', async () => {
+    const layoutName = 'layout-name-deletion';
+    const content = [
+      {
+        type: 'text',
+        content: 'This are the text contents of the template for {{firstName}}',
+        styles: { textAlign: 'left' },
+      },
+      {
+        type: 'button',
+        content: 'SIGN UP',
+        url: 'https://url-of-app.com/{{urlVariable}}',
+      },
+    ];
+    const variables = [
+      { name: 'firstName', type: TemplateVariableTypeEnum.STRING, defaultValue: 'John', required: false },
+    ];
+    const isDefault = true;
+    const response = await session.testAgent.post(BASE_PATH).send({
+      name: layoutName,
+      content,
+      variables,
+      isDefault,
+    });
```

Comment:
Shouldn't we create it from the repository level instead of calling the API? Because if anything is changed in the `post` method, then this test will also require an update. WDYT?

Reply:
My take is that for E2E testing to try to set up everything at API level. If any of the endpoints would change, being used in the different tests will help to confirm if the changes made are retro-compatible. For that point, and peace of mind of not introducing breaking changes, I prefer this approach than use the repository layer to create the expectations needed for the tests.